_docs_custom_metrics/generic-script/index.md

```diff
@@ -4,128 +4,174 @@ title: Generic script plugin
+
+When the agent is started, it will execute `<script> -c` to retrieve the metric and events defined by the script. The output should be a json that contains the metrics and events.
+
+{% include alert.html type="warning" text="`<script> -c` should return within 1 second, otherwise it will be killed. The script should return exit code 0. Otherwise, the output will not be parsed and an error will be logged." %}
 
 This is an **example** output of a script in **configuration mode**:
 
 {% highlight json %}
 {
-    "maxruntime": 1000,
+    "maxruntime": 5000,
     "period": 60,
     "metrics": [{
         "id": 1,
-        "datatype": "DOUBLE",
-        "name": "First metric",
-        "description": "Description for first metric",
-        "groups": "MyMetrics/Category1",
-        "unit": "Bytes",
-        "tags": "MYTAG1,MYTAG2",
-        "calctype": "Difference",
+        "name": "Incoming traffic on interface",
+        "description": "Incoming traffic in bytes on network interface",
+        "groups": "Networking/Traffic",
+        "unit": "b/s",
+        "tags": "NETWORK",
+        "datatype": "GUAGE",
+        "calctype": "Instant",
         "dimensions": [
-            {"id": 1, "name":"dimension1"},
-            {"id": 2, "name":"dimension2"},
-            {"id": 3, "name":"dimension3"}
+            {"id": 1, "name":"interface"}
         ]
     }, {
         "id": 2,
-        "datatype": "DOUBLE",
         "name": "Second metric",
         "description": "Description for second metric",
-        "groups": "MyMetrics/Category1",
+        "groups": "MyMetrics/Category2",
         "unit": "",
         "tags": "MYTAG1,MYTAG2",
-        "calctype": "Instant",
+        "datatype": "DOUBLE",
+        "calctype": "Difference",
         "dimensions": [
-            {"id": 4, "name":"dimension4"},
-            {"id": 5, "name":"dimension5"}
+            {"id": 2, "name":"dimension2"},
+            {"id": 3, "name":"dimension3"}
         ]
     }],
     "events": [{
         "id": 1,
-        "name": "First event",
-        "description": "Description for first event",
+        "name": "My custom event",
+        "description": "Description for my custom event",
         "attributeDescriptions": [
-            {"name":"firstAttribute", "type":"integer", "unit":"#"}
+            {"name":"Exit code", "type":"integer", "unit":""}
         ]
     }]
 }
 {% endhighlight %}
 
-#### Parameters
+#### maxruntime
+
+The maximum amount of time (in milliseconds) the script is allowed to run in data retrieval mode, if the runtime exceeds this value the script will be stopped. The maximum value for maxruntime is 50000 (50 seconds).
+
+#### period
 
-| Parameter               | Description                                                                                                                                          |
+Time interval (in seconds) between agent polls that retrieve the data. The minimum value for period is 60 (1 minute).
```

Comment:
Minimum value is 60 seconds, and we kill the scripts after 60 seconds. Maybe we should leave this parameter out for now? Not sure if it's required or defaults to 60.

Reply:
Yes, I would prefer to leave this out, but since this release it is required, I would like to make the docs correct.